### PR TITLE
imagebuildah: handle ID mappings for COPY --from

### PR DESCRIPTION
**Pull request by nalind**
Fix handling of ID mapping for COPY: when copying from other containers, use their mappings, and when copying from the host, use host mappings.

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -105,19 +110,108 @@ func convertRuntimeIDMaps(UIDMap, GIDMap []rspec.LinuxIDMapping) ([]idtools.IDMa
...
+		tarIDMappingOptions = &IDMappingOptions{
+			HostUIDMapping: true,
+			HostGIDMapping: true,
+		}
+	}
+	return func(src, dest string) error {
+		logrus.Debugf("copyFileWithTar(%s, %s)", src, dest)
+		f, err := os.Open(src)
+		if err != nil {
+			return errors.Wrapf(err, "error opening %q to digest its contents")
```
Comment (rhatdan) [inline util.go]: Missing input for %q

File: util.go [hunk 2]
```diff
@@ -105,19 +110,108 @@ func convertRuntimeIDMaps(UIDMap, GIDMap []rspec.LinuxIDMapping) ([]idtools.IDMa
...
+		}
+		defer func() {
+			if f != nil {
+				f.Close()
+			}
+		}()
+
+		sysfi, err := system.Stat(src)
+		if err != nil {
+			return errors.Wrapf(err, "error reading attributes of %q")
```
Comment (rhatdan) [inline util.go]: Missing input for %q

File: util.go [hunk 3]
```diff
@@ -105,19 +110,108 @@ func convertRuntimeIDMaps(UIDMap, GIDMap []rspec.LinuxIDMapping) ([]idtools.IDMa
...
+		hostUID := sysfi.UID()
+		hostGID := sysfi.GID()
+		containerUID, containerGID, err := util.GetContainerIDs(tarIDMappingOptions.UIDMap, tarIDMappingOptions.GIDMap, hostUID, hostGID)
+		if err != nil {
+			return errors.Wrapf(err, "error mapping owner IDs of %q: %d/%d", src, hostUID, hostGID)
+		}
+
+		fi, err := os.Lstat(src)
+		if err != nil {
+			return errors.Wrapf(err, "error reading attributes of %q")
```
Comment (rhatdan) [inline util.go]: Bad %q

File: util.go (hunk 1 above)
Comment (nalind) [reply util.go]: Argh, fixed.

**Comment by rhatdan** [comment]
LGTM
@TomSweeneyRedHat @giuseppe @vrothberg @QiWang19 PTAL

**Pull request closed without merging**